sklearn/utils/estimator_checks.py

```diff
@@ -2591,14 +2555,10 @@ def check_estimators_data_not_an_array(name, estimator_orig, X, y, obj_type):
 
 
 def check_parameters_default_constructible(name, Estimator):
-    # this check works on classes, not instances
     # test default-constructibility
     # get rid of deprecation warnings
-    if isinstance(Estimator, BaseEstimator):
-        # Convert estimator instance to its class
-        # TODO: Always convert to class in 0.24, because check_estimator() will
-        # only accept instances, not classes
-        Estimator = Estimator.__class__
+
+    Estimator = Estimator.__class__
```

Comment:
This would actually fail in mypy. The reason it doesn't currently is that `conda activate` fails the the CI step silently is marked as successful. So the linting CI job is broken. Fix in https://github.com/scikit-learn/scikit-learn/pull/17177

Reply:
Are you sure?

`mypy --ignore-missing-import sklearn` passes one this branch (v 0.770)

I've had mypy and linting issues on PRs recently so it seemed to "work" fine

Reply:
and the mypy CI seems to be happy and it was run according to the output "Success: no issues found in 472 source files"

https://dev.azure.com/scikit-learn/scikit-learn/_build/results?buildId=16415&view=logs&jobId=32e2e1bb-a28f-5b18-6cfc-3f01273f5609&j=32e2e1bb-a28f-5b18-6cfc-3f01273f5609&t=ac767730-b8ef-5c12-a6ad-59242901cb7c

Reply:
Ahh, yes you are right, it just installs everything in the main env. Well I would have though that it would error but nevermind.

Reply:
We still need to fix the activation issues in https://github.com/scikit-learn/scikit-learn/pull/17177 though.

I though it was broken because I keep getting a few new mypy errors with code on master (that I haven't touched) occasionally.